src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -484,21 +480,27 @@ public OpenIdConnectAuthenticationHandler(HttpClient backchannel)
                 return null;
             }
 
+            message = authorizationCodeRedeemedContext.ProtocolMessage;
+
             // no need to validate signature when token is received using "code flow" as per spec [http://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation].
             var validationParameters = Options.TokenValidationParameters.Clone();
             validationParameters.ValidateSignature = false;
 
-            ticket = ValidateToken(idToken, message, properties, validationParameters, out jwt);
+            ticket = ValidateToken(message.IdToken, message, properties, validationParameters, out jwt);
```

Comment:
This line is bugged: `message` actually represents the authorization response, not the token response. And since we're in a path that only supports the code flow (and not the hybrid flow), `message.IdToken` is always null. You must instead extract the `id_token` from the token response.

> error   : [Microsoft.AspNet.Authentication.OpenIdConnect.OpenIdConnectAuthentica
tionMiddleware] OIDCH_0017: Exception occurred while processing message.
System.ArgumentNullException: La valeur ne peut pas être null.
Nom du paramètre : IDX10000: The parameter 'tokenstring' cannot be a 'null' or a
n empty string.
   à Microsoft.IdentityModel.Logging.LogHelper.Throw(String message, Type except
ionType, EventLevel logLevel, Exception innerException)
   à System.IdentityModel.Tokens.Jwt.JwtSecurityTokenHandler.CanReadToken(String
 tokenString)
   à Microsoft.AspNet.Authentication.OpenIdConnect.OpenIdConnectAuthenticationHa
ndler.ValidateToken(String idToken, OpenIdConnectMessage message, Authentication
Properties properties, TokenValidationParameters validationParameters, JwtSecuri
tyToken& jwt)